ui/packages/catalog-ui-search/src/main/webapp/js/widgets/drawing.controller.js

```diff
@@ -0,0 +1,147 @@
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+/*global define, require, module*/
+var Marionette = require('marionette');
+var wreqr = require('wreqr');
+
+module.exports = Marionette.Controller.extend({
+    enabled: true,
+    drawingType: 'extendme!',
+    initialize: function() {
+        this.listenTo(wreqr.vent, `search:${this.drawingType}display`, function(model) {
+            this.show(model);
+        });
+        this.listenTo(wreqr.vent, `search:draw${this.drawingType}`, function(model) {
+            this.draw(model);
+        });
+        this.listenTo(wreqr.vent, 'search:drawstop', function(model) {
+            this.stop(model);
+        });
+        this.listenTo(wreqr.vent, 'search:drawend', function(model) {
+            this.destroyByModel(model);
+        });
+        this.listenTo(wreqr.vent, 'search:destroyAllDraw', function(model) {
+            this.destroyAll(model);
+        });
+    },
+    views: [],
+    destroyAll: function() {
+        for (var i = this.views.length - 1; i >= 0; i -= 1) {
+            this.destroyView(this.views[i]);
+        }
+    },
+    getViewForModel: function(model) {
+        return this.views.filter(function(view) {
+            return view.model === model && view.options.map === this.options.map;
+        }.bind(this))[0];
+    },
+    removeViewForModel: function(model) {
+        var view = this.getViewForModel(model);
+        if (view) {
+            this.views.splice(this.views.indexOf(view), 1);
+        }
+    },
+    removeView: function(view) {
+        this.views.splice(this.views.indexOf(view), 1);
+    },
+    addView: function(view) {
+        this.views.push(view);
+    },
+    show: function(model) {
+        if (this.enabled) {
+            var bboxModel = model || new Draw.BboxModel();
+
+            var existingView = this.getViewForModel(model);
+            if (existingView) {
+                existingView.drawStop();
+                existingView.destroyPrimitive();
+                existingView.updatePrimitive(model);
+            } else {
+                var view = new Draw.BboxView({
+                    map: this.options.map,
+                    model: bboxModel
+                });
+                view.updatePrimitive(model);
+                this.addView(view);
+            }
+
+            return bboxModel;
+        }
+    },
+    draw: function(model) {
```

Comment:
❓ This just looks like the draw function from `cesium.bbox.js`/`openlayers.bbox.js`.

Is the idea that this falls back to a bbox draw tool if nothing else is specified? Is there anything else we can do to make it more apparent that this should be overridden? Or can we at least have `cesium.bbox.js` and `openlayers.bbox.js` properly inherit this instead of redefining the function again?

Reply:
Yeah, I'm thinking we should update this to throw an error that says to implement the method.